Gems/ROS2/Code/Tests/UrdfParserTest.cpp

```diff
@@ -846,46 +859,147 @@ namespace UnitTest
         ASSERT_TRUE(AZStd::ranges::contains(joints, "joint1", jointToNameProjection));
     }
 
-    TEST_F(UrdfParserTest, TestPathResolvementGlobal)
+    TEST_F(UrdfParserTest, TestPathResolve_ValidAbsolutePath_ResolvesCorrectly)
     {
-        constexpr AZ::IO::PathView dae = "file:///home/foo/ros_ws/install/foo_robot/meshes/bar.dae";
+        // Verify that an absolute path that wouldn't be resolved by prefixes or ancestor paths 
+        // or the AMENT_PREFIX_PATH will still resolve correctly as long as the absolute path exists on disk.
+        constexpr AZ::IO::PathView dae = "file:///usr/ros/humble/meshes/bar.dae";
         constexpr AZ::IO::PathView urdf = "/home/foo/ros_ws/install/foo_robot/foo_robot.urdf";
-        auto result = ROS2::Utils::ResolveURDFPath(
+        constexpr AZ::IO::PathView expectedResult = "/usr/ros/humble/meshes/bar.dae";
+        auto result = ROS2::Utils::ResolveAssetPath(
             dae,
-            urdf, "",
-            [](const AZ::IO::PathView&) -> bool
+            urdf, "", GetTestSettings(),
+            [expectedResult](const AZ::IO::PathView& p) -> bool
             {
+                // Only a file name that matches the expected result will return that it exists on disk.
+                return p.LexicallyNormal() == expectedResult;
+            });
+        EXPECT_EQ(result.LexicallyNormal(), expectedResult);
```

Comment:
There shouldn't be a need to lexically normal any of these paths.
When comparing `AZ::IO::Path*` classes normalization is not needed.

Normalization is only needed when performing string comparison, such when storing the path into an AZStd::string/AZStd::fixed_string.

Reply:
The unit tests would seem to disagree. I can remove it in most places, but the following is NOT seen as equivalent:
    `/home/foo/ros_ws/install/foo_robot/./meshes/bar.dae`
    `/home/foo/ros_ws/install/foo_robot/meshes/bar.dae`

I'll remove LexicallyNormal() everywhere except for the one test that requires it though.

Reply:
Yeah if those aren't comparing equal, then that is a bug in the AZ::IO::Path [compare relative path code](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/PathParser.inl#L689-L707)

Reply:
I'll fix it,

Reply:
Ty! Here's a unit test that fails for testing the fix:
```
    TEST_F(PathFixture, OperatorEquals_PathAndPathViewCompareSuccessfullyWithLexicallyEqualPaths)
    {
        AZ::IO::Path path1 = "/home/foo/ros_ws/install/foo_robot/./meshes/bar.dae";
        AZ::IO::PathView path2 = "/home/foo/ros_ws/install/foo_robot/meshes/bar.dae";
        EXPECT_EQ(path1, path2);
    }
```

Reply:
Thanks, I'll have a fix in a couple of minutes.

Reply:
Here is the PR with the fix.
https://github.com/o3de/o3de/pull/16674

Reply:
Removed the last LexicallyNormal() calls, thanks!